### PR TITLE
sim: Increase test coverage for ram-load

### PR DESCRIPTION
**Pull request by Roolli**
This PR adds new test cases to the simulator to test the ram-load feature of MCUboot. New test variations have also been added for the simulator CI.

### CODE REVIEW

File: sim/src/image.rs [hunk 1]
```diff
@@ -1269,6 +1329,28 @@ impl Images {
...
     }
 
+    pub fn run_ram_load_boot_with_result(&self, expected_result: bool) -> bool {
+        if !Caps::RamLoad.present() {
+            return false;
+        }
+        // Clone the flash so we can tell if unchanged.
+        let mut flash = self.flash.clone();
+
+        // Create RAM config where the image would not fit.
```
Comment (davidvincze) [inline sim/src/image.rs:1339]: Is it still valid?

File: sim/src/image.rs [hunk 2]
```diff
@@ -221,11 +234,58 @@ impl ImagesBuilder {
...
+        Images {
+            flash,
+            areadesc: self.areadesc,
+            images,
+            total_count: None,
+            ram: self.ram,
+        }
+    }
+
+    pub fn make_no_upgrade_image_manipulated(self, deps:&DepTest, img_manipulation: ImageManipulation) -> Images {
```
Comment (davidvincze) [inline sim/src/image.rs:258]: Could it be combined with make_no_upgrade_image()?

**Review by davidvincze — COMMENTED** [review]
What's the impact of the new test cases on the execution time?

File: sim/src/image.rs (hunk 1 above)
Comment (davidvincze) [reply sim/src/image.rs]: Thanks for the update.

File: sim/src/image.rs (hunk 2 above)
Comment (davidvincze) [reply sim/src/image.rs]: Thanks.

File: .github/workflows/sim.yaml [hunk 3]
```diff
@@ -43,6 +43,10 @@ jobs:
         - "sig-rsa validate-primary-slot direct-xip"
         - "sig-rsa validate-primary-slot ram-load multiimage"
         - "sig-rsa validate-primary-slot direct-xip multiimage"
+        - "ram-load enc-aes256-kw multiimage"
+        - "ram-load enc-aes256-kw sig-ecdsa-mbedtls multiimage"
+        - "ram-load enc-aes256-ec256 multiimage"
+        - "ram-load enc-aes256-x25519 multiimage"
```
Comment (davidvincze) [inline .github/workflows/sim.yaml:49]: I think these 2 can be removed. These image encryption algs are already tested separately and we have
other feature sets to test ram-load with multiple encrypted images. I believe these don't have too much added value.
What do you think?

**Review by davidvincze — COMMENTED** [review]
Please rebase this patch stack.

File: .github/workflows/sim.yaml (hunk 3 above)
Comment (Roolli) [reply .github/workflows/sim.yaml:49]: I have no objections.

**Comment by Roolli** [comment]
> What's the impact of the new test cases on the execution time?

Insignificant, as the tests that are taking a long time aren't being ran.

**Review by davidvincze — APPROVED** [review]

**Pull request merged**